base/compiler/ssair/slot2ssa.jl

```diff
@@ -147,6 +148,7 @@ function fixemup!(cond, rename, ir::IRCode, ci::CodeInfo, idx::Int, @nospecializ
                     return true
                 end
             end
+            # temporarily corrupt the isdefined node. type_lift_pass! will fix it
```

Comment:
Should this be a separate Expr head then?

Reply:
Probably would be more accurate? But possibly annoying to implement? Dunno. I was going for the minimal fix